### DISCUSSION TITLE
Set Data Directoy to OneDrive/cloud storage?

### DISCUSSION THREAD

**Discussion by Tortolonch** [TriliumNext/Trilium#2583] [Q&A] [answered]
As the title says, can I set my data directory to my OneDrive folder as a haphazard way to set up sync?
I'm loving Trilium! It's actually got a bunch of features that I had come up with before, so I'm glad i'm not the only one that thought a notes app should have that functionality. 

I'm a university student and have been using it for almost two weeks. However, I completely forgot that we'll be going back to in-person classes starting next week, and without sync, I'd essentially be leaving the notebooks I started at home on my desktop. I'm not exactly a power user, but I gave it a shot at setting up a server instance using Docker on my Windows desktop, but no luck. At this point, I'm ready to just save my document.db file to a flash drive before leaving the house, and transferring it over to my teeny tiny laptop that barely fits my hands.

While I would have loved being able to implement the server to get browser and mobile usage of my notes, if I can just get it working on my laptop, it's a win in my book.

**Comment by rebron1900** [comment, marked as answer]
https://github.com/zadam/trilium/wiki/FAQ#can-i-use-dropbox--google-drive--onedrive-to-sync-data-across-multiple-computers

**Reply by rebron1900** [reply]
If you need synchronization, use the server version

**Reply by rebron1900** [reply]
You can also synchronize database files via Onedrive, but some unexpected errors may occur later

**Reply by Tortolonch** [reply]
I appreciate the response! I spent most of my morning trying to get things to work, that was kind of my nuclear option. I tried to dig around as best as I could to find if my question was already answered here, but I completely missed the FAQ, thank you!

**Reply by rebron1900** [reply]
我最开始是使用oendrive同步数据，使用了大概两个星期，好像没什么大问题，不过使用这个方式同步数据后就不能再使用服务器版本了

I started using Oendrive for about two weeks. It seems to be no big deal, but I can't use the server version after using oendrive this way

**Comment by veritatis** [comment]
Actually it is doable, although not recommended. I'd used trilium this way for quite a while before I finally set up the server on my NAS.

You can just put the trilium folder in onedrive folder and use the portable version of trilium (run "trilium-portable.bat" instead of "trilum.exe"), then all data file will be stored in the "trilium-data" subfolder. The potential trouble is that onedrive won't sync the data file while it is in use (i.e., trilium app is running), so remember to close the trilium application and wait until onedrive finishes uploading the new version of your data file before you shut down windows. And, when you want to sync trilium on another computer, wait until onedrive finish syncing files before you open trilium app, otherwise onedrive will try to keep different version of data files by renaming them and generates lots of rubbish. Never let trilium be opened simultaneously on multiple computers that are synced by onedrive.

Alternatively, you can use symbol link function to sync the data files, but I don't see any benefit compared to just using the portable version.
